analyzers/tests/SonarAnalyzer.UnitTest/TestFramework/Verifier.cs

```diff
@@ -298,16 +405,60 @@ static IEnumerable<TMessage> ReadProtobuf(string path)
                                            OutputKind outputKind,
                                            IEnumerable<MetadataReference> additionalReferences,
                                            string sonarProjectConfigPath = null)
+        {
+            using var scope = new EnvironmentVariableScope(true) { EnableConcurrentAnalysis = true};
+            CreateConcurrencyTest(paths, out var pathsWithConcurrencyTests);
+            var solution = SolutionBuilder.CreateSolutionFromPaths(pathsWithConcurrencyTests, outputKind, additionalReferences);
+            CompileAndVerifyAnalyzer(solution, diagnosticAnalyzers, options, checkMode, sonarProjectConfigPath);
+        }
+
+        private static void CreateConcurrencyTest(IEnumerable<string> paths, out List<string> resultingPaths)
+        {
+            resultingPaths = new List<string>(paths);
+            var language = AnalyzerLanguage.FromPath(paths.First());
+            foreach (var path in paths)
+            {
+                var fullName = new FileInfo(path).FullName;
+                var content = File.ReadAllText(fullName, Encoding.UTF8);
+                if (language == AnalyzerLanguage.CSharp)
+                {
+                    var newPath = $"{fullName}.cs";
+                    File.WriteAllText(newPath, $"namespace AppendedNamespaceForConcurrencyTest {{ {content} }}");
+                    resultingPaths.Add(newPath);
+                }
+                else
+                {
+                    var newPath = $"{fullName}.vb";
+                    File.WriteAllText(newPath, InsertNamespaceForVB(content));
+                    resultingPaths.Add(newPath);
+                }
+            }
+        }
+
+        private static string InsertNamespaceForVB(string content)
+        {
+            var match = Regex.Match(content, @"^\s*Imports\s+.+$", RegexOptions.Multiline | RegexOptions.RightToLeft);
+            var idx = match.Success ? match.Index + match.Length + 1 : 0;
+            return content.Insert(idx, "Namespace AppendedNamespaceForConcurrencyTest\n") + "\nEnd Namespace";
```

Comment:
While C# version preserves line numbers, this messes them up for VB. This should do the trick and maybe re-enable some affected test cases.

```suggestion
            return content.Insert(idx, "Namespace AppendedNamespaceForConcurrencyTest : ") + " : End Namespace";
```

Reply:
I don't think that we've disabled any VB case regarding that one. But added it for the simplicitly to detect the locations in original files